cpp/oneapi/dal/algo/kmeans/test/spmd_backend_fixture.hpp

```diff
@@ -0,0 +1,669 @@
+            std::numeric_limits<float_t>::epsilon() * iteration_count * 10;
+        check_centroid_match_with_rel_tol(match_map, strict_rel_tol, ref_centroids, centroids);
+        check_response_match(match_map, ref_responses, responses);
+
+        if (test_convergence) {
+            INFO("check convergence");
+            REQUIRE(iteration_count < desc.get_max_iteration_count());
+        }
+    }
+
+    bool check_value_with_ref_tol(float_t val, float_t ref_val, float_t ref_tol) {
+        float_t max_abs = std::max(fabs(val), fabs(ref_val));
+        if (max_abs == 0.0)
+            return true;
+        CAPTURE(val, ref_val, fabs(val - ref_val) / max_abs, ref_tol);
+        return fabs(val - ref_val) / max_abs < ref_tol;
+    }
+
+    void check_base_infer_result(const descriptor_t& desc,
+                                 const infer_result_t& result,
+                                 float_t ref_objective_function) {
+        const auto [responses, objective_function] = unpack_result(result);
+
+        check_nans(result);
+
+        INFO("check if non-negative objective function value is expected")
+        REQUIRE(objective_function >= 0.0);
+
+        float_t rel_tol = 1.0e-5;
+        if (!(ref_objective_function < 0.0)) {
+            CAPTURE(objective_function, ref_objective_function, rel_tol);
+            REQUIRE(check_value_with_ref_tol(objective_function, ref_objective_function, rel_tol));
+        }
+    }
+
+    void check_infer_result(const descriptor_t& desc,
+                            const infer_result_t& result,
+                            const table& ref_responses,
+                            float_t ref_objective_function) {
+        const auto [responses, objective_function] = unpack_result(result);
+
+        check_base_infer_result(desc, result, ref_objective_function);
+        check_response_match(ref_responses, responses);
+    }
+
+    void check_infer_result(const descriptor_t& desc,
+                            const infer_result_t& result,
+                            const array<float_t>& match_map,
+                            const table& ref_responses,
+                            float_t ref_objective_function) {
+        const auto [responses, objective_function] = unpack_result(result);
+        check_base_infer_result(desc, result, ref_objective_function);
+        check_response_match(match_map, ref_responses, responses);
+    }
+
+    void check_centroid_match_with_rel_tol(float_t rel_tol, const table& left, const table& right) {
+        INFO("check if centroid shape is expected")
+        REQUIRE(left.get_row_count() == right.get_row_count());
+        REQUIRE(left.get_column_count() == right.get_column_count());
+
+        INFO("check if centroid match is expected")
+        const auto left_rows = row_accessor<const float_t>(left).pull();
+        const auto right_rows = row_accessor<const float_t>(right).pull();
+        const float_t alpha = std::numeric_limits<float_t>::min();
```

Comment:
```suggestion
        constexpr auto alpha = std::numeric_limits<float_t>::min();
```

Reply:
Let's postpone it in order to keep similar code in all k-means fixtures